src/chains.rs

```diff
@@ -731,22 +797,12 @@ impl<'a> ChainFormatter for ChainFormatterBlock<'a> {
     }
 
     fn child_shape(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<Shape> {
-        Some(
-            if self.root_ends_with_block {
-                shape.block_indent(0)
-            } else {
-                shape.block_indent(context.config.tab_spaces())
-            }
-            .with_max_width(context.config),
-        )
+        let block_end = self.root_ends_with_block;
+        Some(get_block_child_shape(block_end, context, shape))
```

Comment:
This is an example of case where it may be unclear as to why this is being pulled out into a separate function given that it only has this one call site. That's true at the moment, but the number of call sites will expand in future PRs and I've opted to go ahead and pull out here as part of an effort to reduce the diff size (however sleight) of those future PRs

Reply:
That seems very reasonable to me